### PR TITLE
Add docs for gameobjects/particles

### PR DESCRIPTION
**Pull request by samme**
This PR

* Updates the Documentation

### CODE REVIEW

File: src/gameobjects/particles/EmitterOp.js [hunk 1]
```diff
@@ -11,26 +11,95 @@ var GetFastValue = require('../../utils/object/GetFastValue');
...
+ * @param {string} key - The name of the property.
+ * @param {float} t - The normalized lifetime of the particle, between 0 (start) and 1 (end).
+ * @param {number} value - The current value of the property.
+ *
+ * @return {number} The new value of the property.
+ */
+
+/**
+ * Defines a value or an operation yielding a value, setting what a particle property will be at the START of the particle's life, on emit.
+ * @typedef {float|float[]|EmitterOpOnEmitCallback|EmitterOpRandomConfig|EmitterOpRandomMinMaxConfig|EmitterOpRandomStartEndConfig|EmitterOpSteppedConfig|EmitterOpCustomEmitConfig} EmitterOpEmitConfig
```
Comment (samme) [inline src/gameobjects/particles/EmitterOp.js]: Generated: `declare type EmitterOpEmitConfig = ()=>void;`

File: src/gameobjects/particles/EmitterOp.js [hunk 2]
```diff
@@ -11,26 +11,95 @@ var GetFastValue = require('../../utils/object/GetFastValue');
...
+ *
+ * @see Phaser.GameObjects.Particles.Particle#fire
+ */
+
+/**
+ * Defines an operation yielding a value, updating a particle property for the duration of the particle's life.
+ * @typedef {EmitterOpOnUpdateCallback|EmitterOpEaseConfig|EmitterOpCustomUpdateConfig} EmitterOpUpdateConfig
```
Comment (samme) [inline src/gameobjects/particles/EmitterOp.js]: Generated: `declare type EmitterOpUpdateConfig = ()=>void;`

**Comment by samme** [comment]
Two of the generated definitions are wrong. I don't know why.

```typescript
declare type EmitterOpEmitConfig = ()=>void;

declare type EmitterOpUpdateConfig = ()=>void;
```

File: src/gameobjects/particles/EmitterOp.js (hunk 1 above)
Comment (photonstorm) [reply src/gameobjects/particles/EmitterOp.js]: Bloody hell, that's quite a conditional.

**Comment by samme** [comment]
That would be a union alias, I think.

I'll remove those two for now and resubmit.

**Pull request merged**